libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentParser.java

```diff
@@ -111,7 +111,7 @@ public String text() throws IOException {
     }
 
     private void throwOnNoText() {
-        throw new IllegalStateException("Can't get text on a " + currentToken() + " at " + getTokenLocation());
+        throw new IllegalArgumentException("Can't get text on a " + currentToken() + " at " + getTokenLocation());
```

Comment:
nit: I think the message could be improved slightly. Maybe something like "Expected text at LOCATION but found TOKEN"